projects/Douglas Mentoring/day1.py

```diff
@@ -0,0 +1,92 @@
+
+username = input('User Name:')
+password = getpass.getpass()
+
+# Variable for "append" the data later
+formatted_data = []
+
+# for now just to name the file / Soon to add for user to add input :)
+whid = 'YEG1'
+
+
+# Function to sorte the data
+def light_level_check(connection):
+    # Command to checks light level per interface
+    transceiver_output = connection.send_command('show interface transceiver', read_timeout=90,
+                                                 use_textfsm='~/ntc-templates/ntc_templates/templates/cisco_ios_show_interface_transceiver.textfsm')
+
+    for x in transceiver_output:
+        # added both variables below to do a check at that moment based on the "output" for that device interface the loop is going through
+        # why? bc was the way i could think to append(look below) all data to export later in CSV
+        # basically i use the "entry" information from "x" do all the checks and move to the next, and so on.
+        sfpmodel = connection.send_command(f'sh idprom interface {x["iface"]} detail | i Vendor Name')
+        sfpspeed = connection.send_command(
+            f'sh idprom interface {x["iface"]} detail | i Administrative Speed')
+
+        # Checking if that entry maches with 'cisco'
+        if 'CISCO' in sfpmodel.upper():
+            sfpM = 'Cisco'
+        else:
+            sfpM = 'Non-Cisco'
+        # Checking if that entry has a match with 10000, then with 1000
+        if '10000' in sfpspeed:
+            sfpS = '10GB'
+        elif '1000' in sfpspeed:
+            sfpS = '1GB'
+        else:
+            sfpS = 'Unknown'
+        # checking if the light level for fiber connections are acceptable
+        # AND add all that compiled data to "formatted_data"
+        ## I know I could have used "return" but when i did it it was not "appending"
+        ## it was adding all for one device and erasing and add the new ones, should be simple mistake from my end but i could not figure lol
+        if x['rx_pwr'] == 'N/A':
+            formatted_data.append((host, x['iface'], x["rx_pwr"], 'No Connection', sfpM, sfpS))
+        elif float(x['rx_pwr']) >= -10:
+            formatted_data.append((host, x['iface'], x["rx_pwr"], 'Pass', sfpM, sfpS))
+        else:
+            formatted_data.append((host, x['iface'], x["rx_pwr"], 'Fail', sfpM, sfpS))
+
+
+# where script "starts", i could have done a __MAIN__ here but didn't get much how it works yet, currently learning this ;P
+print('Beginning Script')
+
+# Loop to go over device list linked on the beginning
+## I know I can add the connection to a funcion as well, i have it on another script just didn't get to do it as this looks cleaner for me(but i know should be updated)
+for host in file:
+    print(f'Connecting to {host}')
+    try:
+        connection = netmiko.ConnectHandler(host=host, device_type='cisco_ios', username=username,
+                                            password=password, secret=password)
```

Comment:
```suggestion
        connection = netmiko.ConnectHandler(
            host=host, 
            device_type='cisco_ios', 
            username=username,
            password=password, 
            secret=password
        )
```

Reply:
this one i was lazy, i did this way on the test but didn't change here

when first writing script/code for me straight line felt easier

Reply:
All in one line is fine if it fits and you like it.  My team uses 120 character as the maximum line length, as a guide for "fits on one line"